src/tools/rbd_mirror/ImageReplayer.cc

```diff
@@ -1249,6 +1345,57 @@ std::string ImageReplayer<I>::to_string(const State state) {
 }
 
 template <typename I>
+void ImageReplayer<I>::resync_image(Context *on_finish) {
+  dout(20) << dendl;
+
+  bool is_stopped = false;
+  {
+    Mutex::Locker l(m_lock);
+
+    m_stopping_for_resync = true;
+    if (m_state == STATE_STOPPED) {
```

Comment:
I am just concerned about adding more corner cases to ImageReplayer when the user should probably just request the resync via the rbd CLI.